core/src/test/java/com/yahoo/ycsb/workloads/TestGraphWorkload.java

```diff
@@ -0,0 +1,297 @@
+
+    graphWorkload = new GraphWorkload();
+    graphWorkload.init(properties);
+
+    graphWorkload.doTransaction(db, new Object());
+
+    HashSet<String> filenames = new HashSet<>(Arrays.asList(Objects.requireNonNull(new File(directory).list())));
+
+    assertTrue(filenames.contains(Node.NODE_IDENTIFIER + "load.json"));
+    assertTrue(filenames.contains(Edge.EDGE_IDENTIFIER + "load.json"));
+    assertTrue(filenames.contains(Node.NODE_IDENTIFIER + "run.json"));
+    assertTrue(filenames.contains(Edge.EDGE_IDENTIFIER + "run.json"));
+    assertTrue(filenames.contains("nodeIds.txt"));
+    assertTrue(filenames.contains("operations.txt"));
+    assertTrue(filenames.contains("edgeIds.txt"));
+    assertTrue(filenames.contains("componentIds.txt"));
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testModeGenerateDataAndLoadPhaseFailWithRunPhaseOperation() throws WorkloadException {
+    properties.setProperty(Client.DO_TRANSACTIONS_PROPERTY, "false");
+
+    graphWorkload = new GraphWorkload();
+    graphWorkload.init(properties);
+
+    graphWorkload.doTransaction(db, new Object());
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testModeRunBenchmarkAndLoadPhaseFailWithoutNecessaryFilesPresent() throws WorkloadException {
+    properties.setProperty(Client.DO_TRANSACTIONS_PROPERTY, "false");
+
+    graphWorkload = new GraphWorkload();
+    graphWorkload.init(properties);
+
+    graphWorkload.doTransaction(db, new Object());
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testModeRunBenchmarkAndRunPhaseFailWithoutNecessaryFilesPresent() throws WorkloadException {
+    properties.setProperty(Client.DO_TRANSACTIONS_PROPERTY, "false");
+
+    graphWorkload = new GraphWorkload();
+    graphWorkload.init(properties);
+
+    graphWorkload.doTransaction(db, new Object());
+  }
+
+  private void createRunPhaseFiles() throws IOException {
+    new File(directory, Node.NODE_IDENTIFIER + "run.json").createNewFile();
+    new File(directory, Edge.EDGE_IDENTIFIER + "run.json").createNewFile();
+    new File(directory, "nodeIds.txt").createNewFile();
+    new File(directory, "edgeIds.txt").createNewFile();
+    new File(directory, "componentIds.txt").createNewFile();
+    new File(directory, "operations.txt").createNewFile();
+  }
+
+  private void createLoadPhaseFiles() throws IOException {
+    File nodeFile = new File(directory, Node.NODE_IDENTIFIER + "load.json");
+    File edgeFile = new File(directory, Edge.EDGE_IDENTIFIER + "load.json");
+
+    edgeFile.createNewFile();
+    nodeFile.createNewFile();
+  }
+
+  private void createLoadPhaseFilesAndSetLastIds() throws IOException {
+    File nodeFile = new File(directory, Node.NODE_IDENTIFIER + "load.json");
+    File edgeFile = new File(directory, Edge.EDGE_IDENTIFIER + "load.json");
+
+    edgeFile.createNewFile();
+    nodeFile.createNewFile();
+
+    FileWriter fileWriter = new FileWriter(nodeFile);
+    fileWriter.write("{\"id\":{\"" +
```

Comment:
Please, no. Put a file in test resources and access it as a resource, not an explosion in a \" factory.